coral/src/app/features/approvals/acls/components/AclApprovalsTable.test.tsx

```diff
@@ -0,0 +1,263 @@
+        aclRequests={aclRequests}
+        activePage={1}
+        totalPages={10}
+        onApprove={onApprove}
+        onDecline={onDecline}
+        onDetails={onDetails}
+      />
+    );
+    rows = screen.getAllByRole("row");
+    header = rows[0];
+    firstRow = rows[1];
+    secondRow = rows[2];
+  });
+
+  afterEach(cleanup);
+
+  it("describes the table content and pagination for screen readers", () => {
+    screen.getByLabelText(`Acl requests, page 1 of 10`);
+  });
+
+  it("has column to describe the usernames", () => {
+    expect(within(header).getAllByRole("columnheader")[0]).toHaveTextContent(
+      "Principals/Usernames"
+    );
+    expect(within(firstRow).getAllByRole("cell")[0]).toHaveTextContent(
+      "mbasani"
+    );
+  });
+
+  it("has column to describe the ip addresses", () => {
+    expect(within(header).getAllByRole("columnheader")[1]).toHaveTextContent(
+      "IP addresses"
+    );
+    expect(within(secondRow).getAllByRole("cell")[1]).toHaveTextContent(
+      "3.3.3.32 3.3.3.33"
+    );
+  });
+
+  it("has column to describe the topic", () => {
+    expect(within(header).getAllByRole("columnheader")[2]).toHaveTextContent(
+      "Topic"
+    );
+    expect(within(firstRow).getAllByRole("cell")[2]).toHaveTextContent(
+      "aivtopic1"
+    );
+  });
+
+  it("has column to describe the environment", () => {
+    expect(within(header).getAllByRole("columnheader")[3]).toHaveTextContent(
+      "Environment"
+    );
+    expect(within(firstRow).getAllByRole("cell")[3]).toHaveTextContent("1");
+  });
+
+  it("has column to describe the team", () => {
+    expect(within(header).getAllByRole("columnheader")[4]).toHaveTextContent(
+      "Team"
+    );
+    expect(within(firstRow).getAllByRole("cell")[4]).toHaveTextContent("Ospo");
+  });
+
+  it("has column to decsribe the acl type", () => {
+    expect(within(header).getAllByRole("columnheader")[5]).toHaveTextContent(
+      "ACL type"
+    );
+    expect(within(firstRow).getAllByRole("cell")[5]).toHaveTextContent(
+      "CONSUMER"
+    );
+  });
+
+  it("has column to decsribe the status", () => {
+    expect(within(header).getAllByRole("columnheader")[6]).toHaveTextContent(
+      "Status"
+    );
+    expect(within(firstRow).getAllByRole("cell")[6]).toHaveTextContent(
+      "Created"
+    );
+  });
+
+  it("has column to decsribe the author of the request", () => {
+    expect(within(header).getAllByRole("columnheader")[7]).toHaveTextContent(
+      "Requested by"
+    );
+    expect(within(firstRow).getAllByRole("cell")[7]).toHaveTextContent(
+      "amathieu"
+    );
+  });
+
+  it("has column to decsribe the timestamp when the request was made", () => {
+    expect(within(header).getAllByRole("columnheader")[8]).toHaveTextContent(
+      "Requested on"
+    );
+    expect(within(firstRow).getAllByRole("cell")[8]).toHaveTextContent(
+      "06-Jan-2023 14:50:37 UTC"
+    );
+  });
+
+  it("has column for action to view request details", async () => {
```

Comment:
Tiny nitpick note (also goes for the other actions)
-> while the test is testing everything it should, I personally try to use one expect per it (or rather: one specific test case per `it`), so I would do 

```
it "test that button is here" [...] 
const button = within(firstRow).getAllByRole("cell")[9]).getByRole("button", {
        name: "View topic request for aivtopic1",
      })
expect(button).toBeEnabled()
```

and 

```
it "shows detail modal / triggers detail modal when user clicks 'View details'" [...] 
const button = within(firstRow).getAllByRole("cell")[9]).getByRole("button", {
        name: "View topic request for aivtopic1",
      })

await userEvent.click(button)
expect(onDetails).toHaveBeenNthCalledWith(1, aclRequests[0]);
```

Reply:
I thought of splitting it but figured that the `getByRole` would throw anyway if the button was not there so i did not want to repeat it :)

But i guess if the other test checked for enabled and the other for the action it could be worth the split.

Thinking it this way: If someday this test breaks, is it easier to understand:
a.) button is not disabled
b.) fn was not called

in my experience b is usually because of a so realizing that faster could make sense.